### PR TITLE
Reworked error output for ID and SemVer parse errors

### LINKED ISSUE #159: `InvalidId(InvalidId)`
When the client reports an invalid ID error, it should say what the ID is.

### PR DESCRIPTION
**Pull request by technosophos**
This results in error output that walks the nested errors and prints the root cause:

```console
$ bindle --server $BINDLE_HOST push foo
Error: Invalid id: InvalidId("foo")

Caused by:
    Invalid bindle ID foo. A bindle ID should be NAME/VERSION

$ bindle --server $BINDLE_HOST push foo/bar/dadsfas
Error: Invalid id: InvalidSemver("dadsfas")

Caused by:
    ID does not contain a valid semantic version (e.g. 1.2.3): dadsfas
```

Closes #159

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

### CODE REVIEW

File: bin/client/main.rs [hunk 1]
```diff
@@ -26,7 +26,24 @@ mod opts;
 use opts::*;
 
 #[tokio::main]
-async fn main() -> std::result::Result<(), ClientError> {
+async fn main() {
+    // Trap and format error messages using the proper value
+    if let Err(e) = run().await.map_err(|e| anyhow::Error::new(e)) {
+        eprintln!("{}", e);
+        for (i, cause) in e.chain().enumerate() {
+            // Skip the first message
```
Comment (thomastaylor312) [inline bin/client/main.rs:34]: Nit: I forgot to add to this to say we skip it because it is the same as the initial error that gets printed

**Review by thomastaylor312 — APPROVED** [review]
Looks good. One minor thing we forgot in a comment, but no need for another review

File: bin/client/main.rs (hunk 1 above)
Comment (technosophos) [reply bin/client/main.rs]: Fixed

**Comment by technosophos** [comment]
Okay, so I just tested, and hyper remains blissfully oblivious to SSL. So I will need to fix that. Might as well do that in this PR.

**Pull request merged**